src/ripple/rpc/handlers/WalletPropose.cpp

```diff
@@ -31,9 +31,30 @@
 #include <ripple/rpc/handlers/WalletPropose.h>
 #include <ed25519-donna/ed25519.h>
 #include <boost/optional.hpp>
+#include <cmath>
+#include <map>
 
 namespace ripple {
 
+double
+shannon_entropy (std::string const& input)
```

Comment:
Good for you, @seelabs.  With your prod I got off my butt and tried out the function here: http://melpon.org/wandbox/permlink/ccoYfIJP31fXpsYk.  It looks like `shannon_entropy()` does pretty much what we hope for.  A string of 20 unique characters gives an entropy of 86.  A string of identical characters gives a entropy of "-0" :smile:.